sdk/core/azure-core/src/http/policy.cpp

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <http/http.hpp>
+#include <http/policy.hpp>
+
+using namespace Azure::Core::Http;
+
+std::unique_ptr<Response> NextHttpPolicy::ProcessNext(Context& ctx, Request& req)
+{
+  if (m_policies == nullptr)
+    throw;
+
+  if (m_index == m_policies->size() - 1)
+  {
+    //All the policies have run without running a transport policy
+    throw;
+  }
+
+  return (*m_policies)[m_index + 1]->Process(ctx, req, NextHttpPolicy{m_index + 1, m_policies});
```

Comment:
Good solution to how to make this opaque to policies!

Reply:
Kudos goes to Billy on this